spec/record_net_http_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'spec_helper'
+require 'diffy'
+require 'rack'
+require 'rack/handler/webrick'
+
+class HelloWorldApp
+  def call(env)
+    req = Rack::Request.new(env)
+    case req.path_info
+    when /hello/
+      [200, {"Content-Type" => "text/html"}, ["Hello World!"]]
+    when /goodbye/  
+      [500, {"Content-Type" => "text/html"}, ["Goodbye Cruel World!"]]
+    else
+      [404, {"Content-Type" => "text/html"}, ["I'm Lost!"]]
+    end
+  end
+end
+
+describe 'Net::HTTP handler' do
+  include_context 'collect events'
+
+  def get_hello
+    http = Net::HTTP.new('localhost', 19292)
+    http.get '/hello?msg=hi'
```

Comment:
Can we test with multiple query parameters, some duplicated and some urlencoded?